### DISCUSSION TITLE
read from text

### DISCUSSION THREAD

**Discussion by JWDobken** [pyvista/pyvista#2610] [Q&A]
Is it possible to read directly from a text variable?

As a minimal working example I have a simple obj square:

```py
obj_square = """
v 0 0 0
v 5 0 0
v 5 5 0
v 0 5 0
f 1 2 3 4
"""
```

The result I expect is  that I can read this obj text directly:

```py
mesh = pyvista.read(obj_square)
```

What does work if I save and read it as a file:

```py
with open("square.obj", "w") as f:
    f.write(obj_square)
mesh = pyvista.read("square.obj")
cpos = mesh.plot()
```

**Comment by RichardScottOZ** [comment]
You could map it to a file type object in memory perhaps?

**Comment by RichardScottOZ** [comment]
Problem might be that looks like it is checking for a path like thing... a named tempfile?

**Comment by RichardScottOZ** [comment]
Other idea also intermediate, but you could do:-
```py
df = pd.read_csv(StringIO(obj_square), delim_whitespace=True, skipfooter=1, usecols=[i for i in range(1,4)], header=None, names=['x','y','z'])
df_f = pd.read_csv(StringIO(obj_square), delim_whitespace=True, skiprows=5, usecols=[i for i in range(1,5)], header=None).to_numpy()

```
in appropriate format for mesh making if you have to add a column telling it number of faces or vertices etc.

**Comment by banesullivan** [comment]
Take a look at this https://github.com/pyvista/pyvista-support/issues/84#issuecomment-565480216
